frappe/core/doctype/doc_history_temp/doc_history_temp.json

```diff
@@ -0,0 +1,152 @@
+{
```

Comment:
To your first point...I don't think so. Obviously, when testing this out, the user wants to see immediate feedback. But how many actual use cases are there when a user changes something, then immediately needs to see feedback for, as an example, pricing history? I think the short queue is plenty fast for real use as a tradeoff to avoid longer save times. Of course, if you feel otherwise, it's trivial to switch it to make the entry before save() is complete.

There is no UI component to this. There definitely could be reports built into the product in the future, but that's a whole new topic.